ERCS/erc-7579.md

```diff
@@ -48,149 +48,164 @@ The smart account's `validateUserOp` function SHOULD return the return value of
-    * MUST revert if any call was not successful
-    */
-    function executeBatch(Execution[] calldata executions) external returns (bytes memory result);
+     * @dev ERC-4337 executeUserOp according to ERC-4337 v0.7
+     *         This function is intended to be called by ERC-4337 EntryPoint.sol
+     * @param userOp PackedUserOperation struct (see ERC-4337 v0.7+)
+     *
+     * MAY implement this function
+     * MUST ensure adequate authorization control: i.e. onlyEntryPointOrSelf
+     */
+    function executeUserOp(UserOperation calldata userOp) external payable;
+}
+```
+
+The execution mode is a `bytes32` value that is structured as follows:
+
+- callType (1 byte): `0x00` for a single `call`, `0x01` for a batch `call` and `0xff` for `delegatecall`
+- execType (1 byte): `0x00` for a normal execution, `0x01` for a try/catch execution
+- unused (4 bytes): this range is reserved for future standardization
+- modeSelector (4 bytes): an additional mode selector that can be used to create further execution modes
+- modePayload (22 bytes): additional data to be passed
+
+Here is a visual representation of the execution mode:
+
+| CallType | ExecType | Unused  | ModeSelector | ModePayload |
+| -------- | -------- | ------- | ------------ | ----------- |
+| 1 byte   | 1 byte   | 4 bytes | 4 bytes      | 22 bytes    |
+
+#### Account configurations
+
+To comply with this standard, smart accounts MUST implement the account config interface below:
 
+```solidity
+interface IAccountConfig {
     /**
-    * MUST execute a `call` to the target with the provided data and value
-    * MUST only allow installed executors to call this function
-    * MUST revert if the call was not successful
-    */
-    function executeFromExecutor(address target, uint256 value, bytes callData) external returns (bytes memory result);
+     * @dev Initializes the account. Function might be called directly or by a factory
+     * @param data encoded data that can be used during the initialization phase
+     *
+     * MUST revert if the account is already initialized
+     */
+    function initializeAccount(bytes calldata data) external payable;
 
     /**
-    * MUST execute a `call` to all the targets with the provided data and value for each target
-    * MUST only allow installed executors to call this function
-    * MUST revert if any call was not successful
-    */
-    function executeBatchFromExecutor(Execution[] calldata executions) external returns (bytes memory result);
-}
+     * @dev Returns the account id of the smart account
+     * @return accountImplementationId the account id of the smart account
+     *
+     * MUST return a non-empty string
+     * The accountId SHOULD be structured like so:
+     *        "vendorname.accountname.semver"
+     * The id SHOULD be unique across all smart accounts
+     */
+    function accountId() external view returns (string memory accountImplementationId);
```

Comment:
I think also very worth to explain this in Rationale. There were some really important/great points discussed.